File: src/OpenSwaggerSchemaPlugin/JsContracts/OpenSwaggerSchemaDotNetContract.cs
using Microsoft.JSInterop;
using System.Threading.Tasks;

namespace OpenSwaggerSchemaPlugin.JsContracts
{
    public class OpenSwaggerSchemaDotNetContract
    {
        private readonly OpenSwaggerSchemaJsContract _jsContract;

        public OpenSwaggerSchemaDotNetContract(OpenSwaggerSchemaJsContract jsContract)
        {
            _jsContract = jsContract;
        }

        [JSInvokable("HandleMenuActionOpenSwaggerSchema")]
        public Task HandleMenuActionOpenSwaggerSchema()
        {
            _jsContract.OpenFile();
            return Task.CompletedTask;
        }

        [JSInvokable("OnLoadFile")]
        public Task OnLoadFile(string content)
        {
            _jsContract.Log(content);
            return Task.CompletedTask;
        }

        public void SetDotNetReference() {}
    }
}

File: src/OpenSwaggerSchemaPlugin/wwwroot/js/OpenSwaggerSchemaDotNetContract.js
var OpenSwaggerSchemaDotNetContract = (function () {
    return {
        setDotNetReference: function (dotnetContractReference) {
            this.dotnetContractReference = dotnetContractReference;
        },

        onMenuClick: function () {
            this.dotnetContractReference.invokeMethodAsync('HandleMenuActionOpenSwaggerSchema');
        },

        onLoadFile: function (content) {
            this.dotnetContractReference.invokeMethodAsync('OnLoadFile', content);
        }
    };
}());